packages/grafana-ui/src/components/Combobox/Combobox.tsx

```diff
@@ -259,6 +259,11 @@ export const Combobox = <T extends string | number>({
       ? 'search'
       : 'angle-down';
 
+  if (items.length > 100) {
```

Comment:
```suggestion
  if (items.length > 100_000) {
```
I think we can bump it to this at the very least. If loading this many options into Combobox is an issue, it is not related to the component itself. Also, we might want to extract this into a const `RECOMMENDED_ITEMS_AMOUNT`